### ISSUE TITLE
Docker Desktop stuck in running state or crashes after some time on Windows 11 after updating from 4.0.0 to 4.1.0

### ISSUE THREAD

**Issue by nbu** [docker/for-win#12230]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 3B179A18-97D4-40C3-B653-2B72841298F6/20211001182221

### Actual behavior
Docker desktop stuck in starting mode forever or fails with attached logs

### Expected behavior
Docker desktop must start successfully

### Information

  - Windows Version: Windows 11
  - Docker Desktop Version: 4.1.0
  - WSL2 or Hyper-V backend?  WSL2
  - Are you running inside a virtualized Windows e.g. on a cloud server or a VM: No

### Steps to reproduce the behavior

  1. Update windows via Insider Program to the Windows 11 Release
  2. Update Docker Desktop from 4.0.0 to 4.1.0

**Comment by danielwagn3r** [comment]
Seem's to be the same problem as in https://github.com/docker/for-win/issues/12229

**Comment by nbu** [comment]
> Seem's to be the same problem as in #12229

Just checked. I don't have Arch Linux installed. Have Ubuntu, Suse, kali-linux and Debian in WSL

**Comment by nbu** [comment]
Was able to get Docker Desktop 4.1.0 running successfully on Windows 11 by deleting all my WSL distros, installing Docker Desktop. Then, when I have installed the distros again, Docker Desktop kept working.

**Comment by kartas39** [comment]
> Was able to get Docker Desktop 4.1.0 running successfully on Windows 11 by deleting all my WSL distros, then installing Docker Desktop 4.1.0. Then, when I have installed the distros again, Docker Desktop kept working.

same here. I had just Ubuntu WSL. After i unregistered that docker desktop started normally

**Comment by joebone** [comment]
Another here, only wsl distributions are docker, docker-data and Alpine
Alpine (Default)
docker-desktop
docker-desktop-data

as in #12229, the same errors are visible in the %appdata/../local/docker/log.txt file

[13:23:04.218][LinuxWSL2Engine   ][Info   ] msg="(8f4b6f15) d8c2c5e1-cross-distro S<-C d9a6a30b-WslIntegrationService-CSharp GET /check[[STACK]]   at Docker.Engines.WslIntegrationService.<WaitForDistroReadyAsync>d__18.MoveNext() in C:\\workspaces\\master-merge\\src\\github.com\\docker\\pinata\\win\\src\\Docker.Engines\\WslIntegrationService.cs:line 0\n"
[13:23:04.218][LinuxWSL2Engine   ][Info   ] distro Alpine mount service not running after 2m1.9478457s (no socket /run/guest-services/distro-services/alpine.sock)
[13:23:04.219][LinuxWSL2Engine   ][Error  ] msg="(8f4b6f15) d8c2c5e1-cross-distro S->C d9a6a30b-WslIntegrationService-CSharp GET /check (156µs): can't access specified distro mount service: stat /run/guest-services/distro-services/alpine.sock: no such file or directory[[STACK]]main.StartCrossDistroServer.func2({0xc0cf20, 0xc0004c42a0})\n\t/go/src/github.com/docker/pinata/linuxkit/pkg/wsl-bootstrap/cross-distro-service.go:58 +0x412\n"

...
 (59e534f5) 956c042f-WslIntegrationService-CSharp C->S WSLCrossDistroService GET /check
[13:23:09.760][LinuxWSL2Engine   ][Info   ] msg="(59e534f5) d8c2c5e1-cross-distro S<-C 956c042f-WslIntegrationService-CSharp GET /check[[STACK]]   at Docker.Engines.WslIntegrationService.<WaitForDistroReadyAsync>d__18.MoveNext() in C:\\workspaces\\master-merge\\src\\github.com\\docker\\pinata\\win\\src\\Docker.Engines\\WslIntegrationService.cs:line 0\n"
[13:23:09.760][LinuxWSL2Engine   ][Error  ] msg="(59e534f5) d8c2c5e1-cross-distro S->C 956c042f-WslIntegrationService-CSharp GET /check (284.5µs): can't access specified distro mount service: stat /run/guest-services/distro-services/alpine.sock: no such file or directory[[STACK]]main.StartCrossDistroServer.func2({0xc0cf20, 0xc0004c50a0})\n\t/go/src/github.com/docker/pinata/linuxkit/pkg/wsl-bootstrap/cross-distro-service.go:58 +0x412\n"

**Comment by obrobrio2000** [comment]
> Was able to get Docker Desktop 4.1.0 running successfully on Windows 11 by deleting all my WSL distros, then installing Docker Desktop 4.1.0. Then, when I have installed the distros again, Docker Desktop kept working.

Thank you a lot!!!

**Comment by DavieBrown** [comment]
I had this same error, for me, this is what appeared to fix it:

```
net stop winmgmt /y
winmgmt /resetrepository
net start winmgmt /y
```
Restart computer.

Note: I'd recommend if you do choose to go down this route, you fully investigate what this command does / take a backup.

**Comment by nbu** [comment]
Closing this issue since 4.1.1 is released, and I don't have issues updating to 4.1.1 at this point.
